source/ncengine/graphics2/diligent/resource/ResourceTypes.cpp

```diff
@@ -141,54 +145,39 @@ void SetArrayRegion(Diligent::IShaderResourceVariable* variable, std::span<Dilig
 
 void InitializeArray(Diligent::IDeviceContext& context, Diligent::IRenderDevice& device, Diligent::IShaderResourceVariable* variable, uint32_t arraySize, bool transition)
 {
-    auto dummyTexture = asset::TextureWithId
-    {
-        asset::Texture
-        {
-            .width = 1u,
-            .height = 1u,
-            .pixelData = std::vector<unsigned char>{0x1A, 0x2A, 0x3A, 0x4A}
-        },
-        1,
-        asset::AssetSubtype::None
+    const auto dummySubType = asset::AssetSubtype::None;
+    const auto dummyTexture = asset::Texture{
+        .width = 1u,
+        .height = 1u,
+        .pixelData = std::vector<unsigned char>{0x1A, 0x2A, 0x3A, 0x4A}
     };
 
-    auto barriers = std::vector<Diligent::StateTransitionDesc>();
-    barriers.reserve(arraySize);
-
-    auto dummyTextures = std::vector<Diligent::RefCntAutoPtr<Diligent::ITexture>>();
-    auto dummyViews = std::vector<Diligent::IDeviceObject*>();
+    auto subResource = Diligent::TextureSubResData{dummyTexture.pixelData.data(), dummyTexture.width * 4u};
```

Comment:
Does this 4u need to be equivalent to whatever's returned by `GetMinimumDimension`?

Reply:
No, diligent wants the stride in bytes. Width * pixelSize. At some point we'll probably want other pixel sizes (many particles could be grayscale + alpha), but for now we force to 4 channels at conversion time.

Reply:
Although we have a constant for that. I'll update here to use it so we break if/when we change things.